### ISSUE TITLE
Improve preprocessor integration documentation and examples

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#143]
### Description
Users are experiencing confusion about how to properly configure mdbook-lint as an mdBook preprocessor. The relationship between preprocessor mode, configuration options, and output settings is unclear.

### Current Issues

1. **Unclear Configuration**: Users aren't sure what the correct `book.toml` configuration should be
2. **Missing Examples**: No clear examples showing both preprocessor and CLI usage patterns
3. **CI/CD Integration**: No GitHub Actions examples for different use cases

#### 1. Complete book.toml Examples
Show the correct preprocessor configuration:
```toml
# Option 1: Basic preprocessor
[preprocessor.mdbook-lint]

# Option 2: With configuration
[preprocessor.mdbook-lint]
fail-on-warnings = true
```

Clarify:
- When/why to use `command = "mdbook-lint"`
- Whether `renderer = ["html"]` is needed
- If `[output.lint]` configuration is ever needed
- How preprocessor mode interacts with `.mdbook-lint.toml` files

##### GitHub Actions for CLI mode:
```yaml
name: Lint mdBook
on: [push, pull_request]
jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - name: Install mdbook-lint
        run: |
          curl -sSL https://github.com/joshrotenberg/mdbook-lint/releases/latest/download/mdbook-lint-linux-x86_64 -o mdbook-lint
          chmod +x mdbook-lint
      - name: Lint documentation
        run: ./mdbook-lint lint --fail-on-warnings docs/src
```

##### GitHub Actions with mdBook build:
```yaml
name: Build mdBook with Linting
on: [push, pull_request]
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - name: Install mdbook and mdbook-lint
        run: |
          # Install mdbook
          curl -sSL https://github.com/rust-lang/mdBook/releases/download/v0.4.35/mdbook-v0.4.35-x86_64-unknown-linux-gnu.tar.gz | tar -xz
          # Install mdbook-lint
          curl -sSL https://github.com/joshrotenberg/mdbook-lint/releases/latest/download/mdbook-lint-linux-x86_64 -o mdbook-lint
          chmod +x mdbook-lint
          # Add to PATH
          echo "$PWD" >> $GITHUB_PATH
      - name: Build book with linting
        run: mdbook build
```

##### For Technical Documentation with Code Examples:
```toml
# .mdbook-lint.toml
# Disable rules that conflict with shell examples
disabled-rules = ["MDBOOK010"]  # Math detection conflicts with shell variables

# Or configure the rule
[rules.MDBOOK010]
enabled = false  # If you don't use math blocks at all
```

##### For Mixed Documentation Structure:
```toml
# When you have docs outside the book
[mdbook-lint]
orphaned-files-check = false  # Disable MDBOOK005 if needed
```

#### 4. Troubleshooting Guide

**Q: Dollar signs in code blocks trigger errors**
A: This is a known issue with MDBOOK010. Disable it if you have many shell examples.

**Q: Orphaned files warnings for files outside book**
A: MDBOOK005 currently checks beyond the book directory. Use CLI mode with specific paths or disable the rule.

**Q: Should I use preprocessor or CLI mode?**
A: 
- Preprocessor: Integrated with `mdbook build`, catches issues during build
- CLI: Better for CI/CD, more control over what gets linted

### Additional Context
This feedback came from a user who successfully set up mdbook-lint but experienced confusion about the configuration. Clear examples would have saved significant time.

### Acceptance Criteria
- [ ] Add complete book.toml examples to README
- [ ] Create a `docs/preprocessor-setup.md` guide
- [ ] Add GitHub Actions examples to documentation
- [ ] Include troubleshooting section for common issues
- [ ] Show how to handle shell code examples properly
